lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -438,7 +438,7 @@ tokenize([$:, H | T] = Original, Line, Column, Scope, Tokens) when ?is_quote(H)
         true ->
           WarnMsg = io_lib:format(
             "found quoted atom \"~ts\" but the quotes are not required. "
-            "Atoms made exclusively of ASCII letters, numbers, and underscores "
+            "Atoms made exclusively of ASCII letters, numbers, and underscores which may optionally end with ! or ?"
```

Comment:
I like the second option more but I think it needs `ing` in `ending` to form a valid sentence, like this:
```suggestion
            "Atoms made exclusively of ASCII letters, numbers, and underscores, and optionally ending with ! or ? "
```

Reply:
Correct. I was about to suggest the same thing.